### PR TITLE
feat(oauth): add oidc username claim

### LINKED ISSUE #355: ❓ Question: Internal Error while setting the OIDC
### 🙋‍♂️ Question

Hello!

I'm trying to setup SSO on my NAS, using pingvin-share on a docker on said host. I was able to run the SSO Server on my NAS, add an app entry to generate Client ID & Secrets, callback URL and the Wellknown URL too.
Capture d’écran 2023-12-12 à 11 09 45

I was also able to go to the admin panel and fill the corresponding fields:
Screenshot 2023-12-12 at 11-14-12 Paramètres - Pingvin Share

However, the connection fails at when trying to log via SSO, and I don't understand OpenID enought to getwhat is wrong. Server side, the logs says a successful attempts (`"User ••• signed in to application Pinginv via SSO."`), but there is an internal error Client side. It looks like the Username is not managed, or the claim is not able to bind my Pingvin account with my server account (I have redacted sensible part of the log)

```
2023/12/12 01:33:38	stderr	    at async OAuthController.callback (/opt/app/backend/dist/src/oauth/oauth.controller.js:53:13)
2023/12/12 01:33:38	stderr	    at async OAuthService.link (/opt/app/backend/dist/src/oauth/oauth.service.js:77:9)
2023/12/12 01:33:38	stderr	    at zr.request (/opt/app/backend/node_modules/@prisma/client/runtime/library.js:122:7307)
2023/12/12 01:33:38	stderr	    at zr.handleAndLogRequestError (/opt/app/backend/node_modules/@prisma/client/runtime/library.js:122:7697)
2023/12/12 01:33:38	stderr	    at zr.handleRequestError (/opt/app/backend/node_modules/@prisma/client/runtime/library.js:122:8007)
2023/12/12 01:33:38	stderr	    at Ur (/opt/app/backend/node_modules/@prisma/client/runtime/library.js:115:5803)
2023/12/12 01:33:38	stderr	Argument `providerUsername` is missing.
2023/12/12 01:33:38	stderr	
2023/12/12 01:33:38	stderr	     })
2023/12/12 01:33:38	stderr	       }
2023/12/12 01:33:38	stderr	     +   providerUsername: String
2023/12/12 01:33:38	stderr	         providerUserId: "•••",
2023/12/12 01:33:38	stderr	         provider: "oidc",
2023/12/12 01:33:38	stderr	         userId: "•••",
2023/12/12 01:33:38	stderr	       data: {
2023/12/12 01:33:38	stderr	→ 77 await this.prisma.oAuthUser.create({
2023/12/12 01:33:38	stderr	  76 }
2023/12/12 01:33:38	stderr	  75     ]);
2023/12/12 01:33:38	stderr	  74         `provider_${provider}`,
2023/12/12 01:33:38	stderr	
2023/12/12 01:33:38	stderr	/opt/app/backend/dist/src/oauth/oauth.service.js:77:37
2023/12/12 01:33:38	stderr	Invalid `this.prisma.oAuthUser.create()` invocation in
2023/12/12 01:33:38	stderr	PrismaClientValidationError: 
2023/12/12 01:33:38	stderr	Argument `providerUsername` is missing.[39m
2023/12/12 01:33:38	stderr	
2023/12/12 01:33:38	stderr	     })
2023/12/12 01:33:38	stderr	       }
2023/12/12 01:33:38	stderr	     +   providerUsername: String
2023/12/12 01:33:38	stderr	         providerUserId: "•••",
2023/12/12 01:33:38	stderr	         provider: "oidc",
2023/12/12 01:33:38	stderr	         userId: "•••",
2023/12/12 01:33:38	stderr	       data: {
2023/12/12 01:33:38	stderr	→ 77 await this.prisma.oAuthUser.create({
2023/12/12 01:33:38	stderr	  76 }
2023/12/12 01:33:38	stderr	  75     ]);
2023/12/12 01:33:38	stderr	  74         `provider_${provider}`,
2023/12/12 01:33:38	stderr	
2023/12/12 01:33:38	stderr	/opt/app/backend/dist/src/oauth/oauth.service.js:77:37
2023/12/12 01:33:38	stderr	Invalid `this.prisma.oAuthUser.create()` invocation in
2023/12/12 01:33:38	stderr	[31m[Nest] 40  - [39m12/12/2023, 1:33:38 AM [31m  ERROR[39m [38;5;3m[ExceptionsHandler] [39m[31m 
```

According to [this documentation](https://kb.synology.com/en-global/DSM/tutorial/set_up_oidc_for_dsm_in_sso_server#x_anchor_id541dbfc9a1) it looks like the `Authorization scope` and the `Username claim` has to be set Client side.

Can someone help me figure this out?

### PR DESCRIPTION
**Pull request by zz5840**
close #355

### CODE REVIEW

**Comment by stonith404** [comment]
Thanks :) I'm currently not at home but I'll review it at the start of the next week.

File: backend/src/oauth/provider/genericOidc.provider.ts [hunk 1]
```diff
@@ -108,6 +108,7 @@ export abstract class GenericOidcProvider implements OAuthProvider<OidcToken> {
   async getUserInfo(
     token: OAuthToken<OidcToken>,
     query: OAuthCallbackDto,
+    claim: string = undefined,
```
Comment (stonith404) [inline backend/src/oauth/provider/genericOidc.provider.ts:111]:
```suggestion
    claim?: string
```

**Review by stonith404 — APPROVED** [review]
Sorry, I was interrupted while reviewing. Yes this looks fine, thanks!

**Comment by stonith404** [comment]
@zz5840  Can we merge?

**Comment by zz5840** [comment]
@stonith404 Yeah, please merge it.

**Pull request merged**